src/configobj/__init__.py

```diff
@@ -757,7 +758,7 @@ def merge(self, indict, decoupled=False):
         """
         for key, val in list(indict.items()):
             if decoupled:
-                val = copy.deepcopy(val)
+                val = copy(val)
```

Comment:
I tried to pull the `_keyword` regex out of `ConfigObj` and into a separate class. Unsurprisingly, the `deepcopy` still fails.  So, I must have misunderstood what you meant. My apologies, could you rephrase?